front-end/src/app/shared/services/scha-transaction.service.ts

```diff
@@ -12,7 +12,14 @@ export class SchATransactionService {
 
   public get(id: string): Observable<SchATransaction> {
     return this.apiService
-      .get<SchATransaction>(`/sch-a-transactions/${id}`)
+      .get<SchATransaction>(`/sch-a-transactions/${id}/`)
+      .pipe(map((response) => SchATransaction.fromJSON(response)));
+  }
+
+  public getPreviousTransaction(contact_id: string, contribution_date: Date): Observable<SchATransaction> {
+    const contributionDateString: string = contribution_date.toISOString().split('T')[0];
```

Comment:
Maybe consider using DatePipe directly as referenced here: https://stackoverflow.com/questions/43630445/how-to-convert-current-date-to-yyyy-mm-dd-format-with-angular-2#answer-48785621